### PR TITLE
add pod constraint in general estimator

### PR DESCRIPTION
**Pull request by Garrybest**
Signed-off-by: Garrybest <garrybest@foxmail.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Max allowed pod number is an invisible constraint when calculating max available replicas in general estimator.
This PR is related to #723.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/estimator/client/general.go [hunk 1]
```diff
@@ -33,13 +31,17 @@ func (ge *GeneralEstimator) MaxAvailableReplicas(clusters []*clusterv1alpha1.Clu
...
 	resourceSummary := cluster.Status.ResourceSummary
-
 	if resourceSummary == nil {
 		return 0
 	}
 
+	allowedPodNumber := resourceSummary.Allocatable.Pods().Value() - resourceSummary.Allocated.Pods().Value() - resourceSummary.Allocating.Pods().Value()
+	if allowedPodNumber <= 0 {
+		return 0
+	}
```
Comment (RainbowMango) [inline pkg/estimator/client/general.go:42]: How is it possible that the pod < 0? I can imagine `allowedPodNumber==0`.
Shall we add a comment on it?

Comment (Garrybest) [reply pkg/estimator/client/general.go:42]: When too may pods have been created, they cannot been scheduled due to lack of `pod` resource. So the variable `allocating` may be large.

**Comment by RainbowMango** [comment]
LGTM
But you might need to rebase your code as the master just fixed a compile failure.

**Pull request merged**